### PR TITLE
Liberty package enhancement with integration tests

### PR DESCRIPTION
**Pull request by josephca**

### CODE REVIEW

File: docs/libertyPackage.md [hunk 1]
```diff
@@ -2,13 +2,15 @@
...
+| include | Packaging type. Can be used with values `all`, `usr`, `minify`, `wlp`, `runnable`, `all,runnable` and `minify,runnable`. The default value is `all`. The `runnable`, `all,runnable` and `minify,runnable` values are supported beginning with 8.5.5.9 and works with `jar` type archives only. This must be declared in the `packageLiberty` closure. | Yes, only when the `os` option is set. |
+| archive | Location of the target file or directory. If the 'archive' is set to an existing directory, the contents of the server instance will be compressed into `${archive}/${project.name}.zip`&#124;`jar` file. If the target file or directory doesn't exist, the contents of the server instance will be compressed into the specified file, `${buildDir}/${archive}.zip`&#124;`jar`. If the `archive` option is not set, it defaults to `${buildDir}/libs/${project.name}.zip`&#124;`jar`. A jar file is created when the packaging type is either `runnable`,`all,runnable` or `minify,runnable`. A zip file is created for other packaging types. | No |
```
Comment (patricktiu) [inline docs/libertyPackage.md]: I think it is incorrect to say:

`If the target file or directory doesn't exist, the contents of the server instance will be compressed into the specified file, ${buildDir}/${archive}.zip&#124;jar.`

But, you can simply change it to:

`If the target file or directory doesn't exist, the contents of the server instance will be compressed into the specified file.`

File: src/integTest/groovy/net/wasdev/wlp/gradle/plugins/AbstractIntegrationTest.groovy [hunk 2]
```diff
@@ -50,8 +43,8 @@ abstract class AbstractIntegrationTest {
             }
         }
     }
-
-    protected static File createTestProject(File parent, File sourceDir, String buildFilename) {
+    
+    protected static File createTestProject(File parent, File sourceDir) {
```
Comment (patricktiu) [inline src/integTest/groovy/net/wasdev/wlp/gradle/plugins/AbstractIntegrationTest.groovy]: Are you intentionally reverting Dylan's change?  Have you discussed with him?

File: src/integTest/resources/liberty-package-test/liberty-package-archiveJar.gradle [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+            runtimeUrl = wlpUrl
+        }
+        serverName ='libertyOnline'
+    } else {
+        installDir = WLP_DIR
+        serverName = 'libertyOffline'
+    }
+
+    packageLiberty {
+        archive = 'testPackage.jar'
```
Comment (patricktiu) [inline src/integTest/resources/liberty-package-test/liberty-package-archiveJar.gradle]: It is incorrect to set archive to a jar file but without setting include to `runnable` even if there will be not error in the build.

Comment (josephca) [inline src/integTest/resources/liberty-package-test/liberty-package-archiveJar.gradle]: Fixed.

File: docs/libertyPackage.md (hunk 1 above)
Comment (josephca) [inline docs/libertyPackage.md]: Fixed.

File: src/integTest/groovy/net/wasdev/wlp/gradle/plugins/AbstractIntegrationTest.groovy (hunk 2 above)
Comment (josephca) [inline src/integTest/groovy/net/wasdev/wlp/gradle/plugins/AbstractIntegrationTest.groovy]: I haven't discussed with him yet as he's away - I believe 'buildFilename' is not being used anywhere and we can safely remove the redundant parameter. I also cleaned up other codes  where it's referenced.

Comment (dnwarnock) [inline src/integTest/groovy/net/wasdev/wlp/gradle/plugins/AbstractIntegrationTest.groovy]: I think a better fix may be to call `renameBuildFile` within 'createTestProject' to make configuring a test case simpler

Comment (josephca) [inline src/integTest/groovy/net/wasdev/wlp/gradle/plugins/AbstractIntegrationTest.groovy]: I moved 'renameBuildFile' within 'createTestProject'.

**Pull request merged**